src/historywork/PutSnapshotFilesWork.cpp

```diff
@@ -17,8 +17,11 @@ namespace stellar
 
 PutSnapshotFilesWork::PutSnapshotFilesWork(
     Application& app, std::shared_ptr<StateSnapshot> snapshot)
-    : Work(app, fmt::format("update-archives-{:08x}",
-                            snapshot->mLocalState.currentLedger))
+    : Work(app,
+           fmt::format("update-archives-{:08x}",
+                       snapshot->mLocalState.currentLedger),
+           // Each put-snapshot-sequence will retry correctly
+           BasicWork::RETRY_NEVER)
```

Comment:
same thing here: uploads are likely to fail and retrying the whole thing doesn't seem necessary (so "retry a lot" was about right for this)

Reply:
I'm not really sure what you mean: this work just keeps track of uploads to multiple archives. Each upload to an archive is a WorkSequence, which already handles retries in case of a failure, so I don't see a reason to "retry a lot".

Reply:
yes you are right, I missed the nested work.
I think the general strategy seems to be:
* work in general should not retry
* topmost work should retry forever (typically done by rescheduling like how "PublishWork" is scheduled)
* "leaf" work should retry "a lot" (as to make progress)
* work sequences that spawn "leaf" work, may retry once (the reason a sequence like this would have to retry is because of some weird event that "should never happen" that is causing a leaf to be stuck, like a file got deleted on disk)

Now that I see it, the function `BasicWork::getRetryDelay()` allows for timeouts that are way too high: over an hour. We should cap to something more reasonable like up to 1024 seconds (so 10 instead of 12).